### PR TITLE
Space Travelers' Hub v1

### PR DESCRIPTION
**Pull request by AkashaRojee**
Hello!

Together with @moise10r  and @elasfarc, we built the Space Travelers' Hub web app. Please, find the details below and review the project.

Thanks!

## Live Demo

[View live demo of project](https://www.akasharojee.codes/space-travelers-hub)

## Video

[View video about project](https://www.loom.com/share/48a08a3b0de14dbfa6de0594d31e7dd7)

### 1. Setup

- Set up React, Git flow and linters, and customise configurations
- Structure project as feature folders with ducks slices

```
- __tests__
- app //global setup and styling
- common //generic components
- features
  - /feature
    - Feature.js //main file of feature
    - featureSlice.js //feature's reducer
- mocks
```
### 2. Create components

Project is built with React's reusability principle in mind.

#### Hierarchy:

```
- App
  - Header
  - Route: Rockets
    - RDRow(s)
  - Route: Dragons
    - RDRow(s)
  - Route: Missions
    - TableBody
    - TableHeader
  - Route: Profile
    - ProfileColumn(s)
```

- `Rockets` and `Dragons` use a common component `RDRow` to display each row and handle the reserve and cancel actions
- Each column in `Profile` is a reusable component `ProfileColumn`

#### Rockets
- Upon initial render, get list of rockets from rockets API endpoint, save into store, then display in page
- Clicking on Reserve/Cancel button shows/hides Reserved badge and reserves/cancels rocket in store

#### Dragons
- Same as Rockets, but with list of dragons from dragons API endpoint

#### Missions
- Upon initial render, get list of missions from missions API endpoint, save into store, then display in page
- Clicking on Join Mission / Leave Mission button updates participation status to "Active Member"/"NOT A MEMBER", and reserves/cancels the mission in store

#### Profile
- Display reserved rockets, reserved dragons and reserved missions

### 4. Testing

API calls are mocked at network level by using Mock Service Worker, and store is made globally available by customising React Testing Library's render method.

- API handlers and server are in `/mocks`
- Global test setup for API mocking is in `setupTests.js`
- Custom render method is in `test-utils.js`
- Tests are in `/__tests__`

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team,

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,70 +1,89 @@
...
-The page will reload if you make edits.\
-You will also see any lint errors in the console.
+[Live Demo Link](https://AkashaRojee.github.io/space-travelers-hub)
 
-### `npm test`
+## Video
 
-Launches the test runner in the interactive watch mode.\
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+[video link](https://www.loom.com/share/48a08a3b0de14dbfa6de0594d31e7dd7)
```
Comment (mikenath223) [inline README.md:19]: - [x] As stated all members that worked on the project should appear in the video explaining the project. Only 2 members appear. Kindly record a new video with all members.

File: src/features/missions/Missions.js [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+  const handleMission = (id, joined) => {
+    dispatch(joined ? leaveMission(id) : joinMission(id));
+  };
+
+  return (
+    <section className="missions-section">
+      <table className="mission-table">
+        <TableHeader />
+        <TableBody onJoinMission={handleMission} missions={missions} />
+      </table>
```
Comment (mikenath223) [inline src/features/missions/Missions.js:26]: - [ ] OPTIONAL - Nice tests cases, they all pass but an error shows on the terminal: ` Warning: validateDOMNesting(...): <tbody> cannot appear as a child of <div>` Consider fixing it.

**Comment by AkashaRojee** [comment]
Hello, @mikenath223!

Thanks for the review.

Unfortunately, the third member is unreachable.

**Comment by AkashaRojee** [comment]
Hello!

Please note that the third member is unreachable. This is why only 2 members are present in the video.

Thanks!

**Review by oxenprogrammer — APPROVED** [review]
Hi @AkashaRojee , @elasfarc  and @moise10r 

Well done on the project. Since you have explained on the comment that the third member is unreeachable and it being a Friday, I will not hold your progress.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AkashaRojee** [comment]
Hi @oxenprogrammer,

Thanks!

**Comment by mimipeshy** [comment]
### Hello @AkashaRojee  👋,

_This is an additional review as a result of the second opinion request._

It seems like the situation has been resolved, keep up the good work 😄 

Keep rocking 🚀!

**Comment by AkashaRojee** [comment]
Hello, @mimipeshy!

It's been resolved.

Thanks!

**Pull request merged**